src/Autofac/Core/ISharingLifetimeScope.cs

```diff
@@ -50,12 +50,32 @@ public interface ISharingLifetimeScope : ILifetimeScope
         /// <returns><c>true</c> if the key was found; otherwise, <c>false</c>.</returns>
         bool TryGetSharedInstance(Guid id, out object value);
 
+        /// <summary>
+        /// Try to retrieve a shared instance based on a primary GUID key and
+        /// possible secondary qualifying GUID key.
+        /// </summary>
+        /// <param name="primaryId">Key to look up.</param>
+        /// <param name="qualifyingId">Secondary key to look up.</param>
+        /// <param name="value">The instance that has the specified key.</param>
+        /// <returns><c>true</c> if the key was found; otherwise, <c>false</c>.</returns>
+        bool TryGetSharedInstance(Guid primaryId, Guid? qualifyingId, out object value);
```

Comment:
I'm not a super fan of the names, but couldn't thing of anything better. Just trying to take an approach that has minimal external impact to the APIs was my primary focus

Reply:
qualifyingId makes sense to me, but perhaps put more detail into the param comment?

Reply:
I'll see what I can suss out that doesn't sound like word soup :)